internal/config/instance.go

```diff
@@ -185,7 +185,11 @@ func (i *Instance) Get(key string) interface{} {
 		return result
 	}
 	if opt := mediator.GetOption(key); mediator.KnownOption(opt) {
-		return opt.Default
+		default_ := opt.Default
+		if enum, ok := default_.(*mediator.Enums); ok {
+			return enum.Default
+		}
+		return default_
```

Comment:
This ought to be addressed at the type level, or in the mediator. Handling it here sprinkles the business logic for handling these types across three packages.